File: SourceAFIS/Extraction/AbsoluteContrast.cs
using System;
using System.Collections.Generic;
using System.Text;
using SourceAFIS.General;

namespace SourceAFIS.Extraction
{
    public class AbsoluteContrast
    {
        public byte Limit = 15;

        public BinaryMap DetectLowContrast(byte[,] contrast)
        {
            BinaryMap result = new BinaryMap(contrast.GetLength(1), contrast.GetLength(0));
            for (int y = 0; y < result.Height; ++y)
                for (int x = 0; x < result.Width; ++x)
                    if (contrast[y, x] < Limit)
                        result.SetBitOne(x, y);
            Logger.Log(this, result);
            return result;
        }
    }
}

File: java/sourceafis/src/sourceafis/extraction/filters/AbsoluteContrast.java
/**
 * @author Veaceslav Dubenco
 * @since 08.10.2012
 */
package sourceafis.extraction.filters;

import sourceafis.general.BinaryMap;
import sourceafis.general.DetailLogger;
import sourceafis.meta.Parameter;

public final class AbsoluteContrast {
	DetailLogger.Hook logger = DetailLogger.off;

	@Parameter(upper = 255)
	public int Limit = 17;

	public BinaryMap DetectLowContrast(byte[][] contrast) {
		BinaryMap result = new BinaryMap(contrast[0].length, contrast.length);
		for (int y = 0; y < result.getHeight(); ++y)
			for (int x = 0; x < result.getWidth(); ++x)
				if ((contrast[y][x] & 0xFF) < Limit)
					result.SetBitOne(x, y);
		logger.log(result);
		return result;
	}
}
